app/models/enterprise.rb

```diff
@@ -52,9 +54,9 @@ class Enterprise < ApplicationRecord
   belongs_to :owner, class_name: 'Spree::User',
                      inverse_of: :owned_enterprises
   has_many :distributor_payment_methods,
-           inverse_of: :distributor, foreign_key: :distributor_id
+           inverse_of: :distributor, foreign_key: :distributor_id, dependent: :destroy
   has_many :distributor_shipping_methods,
-           inverse_of: :distributor, foreign_key: :distributor_id
+           inverse_of: :distributor, foreign_key: :distributor_id, dependent: :destroy
```

Comment:
If we destroy these relationships, what happens to the payment methods and shipping methods? They could be used by other enterprises or they could be orphaned after this. But even payment methods without enterprise may be referenced by orders. Hm, I think I'm okay with this new behaviour of destroying the link and at some point in the future we may need to clean up the database and remove orphaned records.

Reply:
Updated these to `:retrict_with_exception`